### DISCUSSION TITLE
Document and window

### DISCUSSION THREAD

**Discussion by stephanosn** [idom-team/idom#1222] [Question] [answered]
I have created a css grid where users can enter the number of rows and columns. I want to be able, when the width of the window is smaller than the height, to swap the number of rows and columns around. I can’t seem to find a way to access the window element, so I’ve created a custom event handler that sends the window.innerHeight and width under detail. I receive the event but the detail information doesn’t seem to be there. Any help would be appreciated

**Comment by Archmonger** [comment, marked as answer]
If you need direct access to the client-side DOM, I would recommend switching to `reactpy-django` and use [`pyscript_component`](https://reactive-python.github.io/reactpy-django/latest/reference/components/).

Then, you'd be able to query the DOM within a `use_effect` hook with pyscipt's JS module.

```python
import js

@use_effect
async def peroidic_polling():
    while True:
        if js.window.innerWidth < 1200:
            ...
        await asyncio.sleep(0.5)
```

I am working on migrating pyscript components to core, but a few other things are currently taking priority.
